tests/lib/state.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
```

Comment:
This is included in various pleases, not all use bash. Is bash needed here?

Reply:
Maybe it's just for shellcheck?

Reply:
My point is that while shellcheck will happily check this as bash code it may be sourced from shell (not-bash) context where the special meaning will not work. My question is if we need any bash features for this script.

Reply:
This is not answered.

Reply:
@zyga sh could be used but no wI added [[ "$SPREAD_SYSTEM" = ubuntu-14.04-* ]] and this needs bash, I think this is the only thing that requires bash, otherwise we could use sh.

Reply:
would this work?
```
if [ "${SPREAD_SYSTEM##ubuntu-14.04-*}" = "" ]; then
   # on ubuntu-14.04
   ...
fi
```

Reply:
I think its fine to use bash if it makes things more readable, lets not become purists for no good reason.

Reply:
@mvo5 I left a specific remark, if this code is included from a dash context things will fail without shellcheck warning us about it.

Reply:
I moved the check to the systems.sh, i'll create a new PR to move that script to sh and use that is_ubuntu_14_system check in the other helpers.